isisd/isis_lsp.c

```diff
@@ -1070,6 +1071,80 @@ static struct isis_lsp *lsp_next_frag(uint8_t frag_num, struct isis_lsp *lsp0,
  * Builds the LSP data part. This func creates a new frag whenever
  * area->lsp_frag_threshold is exceeded.
  */
+void iteration_in_lsp_ip(struct isis_extended_ip_reach *r, struct isis_lsp *lsp,
+			 int *count)
+{
+	for (struct isis_item *l = lsp->tlvs->extended_ip_reach.head; l;
+	     l = l->next) {
+		struct isis_extended_ip_reach *rt =
+			(struct isis_extended_ip_reach *)l;
+		if (IPV4_ADDR_SAME(&r->prefix.prefix, &rt->prefix.prefix)) {
+			if ((r->metric > rt->metric) ||
+			    (r->metric == rt->metric))
+				*count += 1;
+			break;
```

Comment:
Here you code terminate the loop once a first occurrence has been founded. Thus, the value of count will be 0 or 1. May be returning a boolean should be sufficient if it is really the goal of this function. If the objective is to count the number of extended IP reachability TLV that have the same IPv4 prefix, you should remove the break instruction.